solr/prometheus-exporter/src/test/org/apache/solr/prometheus/scraper/SolrStandaloneScraperBasicAuthTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.prometheus.scraper;
+
+import io.prometheus.client.Collector;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import org.apache.lucene.tests.util.LuceneTestCase;
+import org.apache.solr.SolrTestCaseJ4;
+import org.apache.solr.client.solrj.impl.Http2SolrClient;
+import org.apache.solr.common.util.ExecutorUtil;
+import org.apache.solr.common.util.IOUtils;
+import org.apache.solr.common.util.SolrNamedThreadFactory;
+import org.apache.solr.prometheus.PrometheusExporterTestBase;
+import org.apache.solr.prometheus.exporter.MetricsConfiguration;
+import org.apache.solr.prometheus.exporter.PrometheusExporterSettings;
+import org.apache.solr.prometheus.exporter.SolrClientFactory;
+import org.apache.solr.prometheus.exporter.SolrScrapeConfiguration;
+import org.apache.solr.prometheus.utils.Helpers;
+import org.apache.solr.util.SolrJettyTestRule;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class SolrStandaloneScraperBasicAuthTest extends SolrTestCaseJ4 {
+
+  @ClassRule public static final SolrJettyTestRule solrRule = new SolrJettyTestRule();
+
+  private static Http2SolrClient solrClient;
+  private static MetricsConfiguration configuration;
+  private static SolrStandaloneScraper solrScraper;
+  private static ExecutorService executor;
+
+  private static String user = "solr";
+  private static String pass = "SolrRocks";
+
+  @BeforeClass
+  public static void setupSolrHome() throws Exception {
+    Path solrHome = LuceneTestCase.createTempDir();
+    Files.copy(
+        SolrTestCaseJ4.TEST_PATH().resolve("security.json"), solrHome.resolve("security.json"));
```

Comment:
a thought...  that if security.json isn't special for your test, just use the one that now ships with Solr?

Reply:
it only has a user+pass. if one already exists and is available to this module I can try to use it, sure

Reply:
trying to find an example for this. would you happen to have one that I could use as a reference?

Reply:
https://github.com/apache/solr/blob/main/solr/core/src/resources/security.json is the one that was introduced...

Reply:
I thin the trouble is prometheus module does not depend on core (not even for testing)? bringing that in just for access to the security file might be overkill.

Reply:
ah.. yeah.

Reply:
found a cleaner way. removed the file and I am recreating it as needed.